lib/index.js

```diff
@@ -40,18 +23,45 @@ S3rver.prototype.resetFs = function (callback) {
   });
 }
 
+S3rver.prototype.getMiddleware =
+S3rver.prototype.callback = function () {
+  return new App(this.options);
+}
+
 S3rver.prototype.run = function (done) {
   const app = new App(this.options);
-  const server = app.serve(done);
-  if (this.options.removeBucketsOnClose) {
-    const close = server.close.bind(server);
-    server.close = (...args) => {
-      this.resetFs(() => {
-        close(...args);
-      });
-    };
-  }
+  let server = ((this.options.key && this.options.cert) || this.options.pfx)
+      ? https.createServer(this.options, app)
+      : app;
+  server = server.listen(this.options.port, this.options.hostname, (err) => {
+    done(err, this.options.hostname, this.options.port, this.options.directory);
+  }).on('error', (err) => {
+    done(err);
+  });
+  server.close = (callback) => {
+    const { close } = Object.getPrototypeOf(server);
+    return close.call(server, () => {
+      app.logger.unhandleExceptions();
+      app.logger.close();
+      if (this.options.removeBucketsOnClose) {
+        this.resetFs(callback);
+      } else {
+        callback();
+      }
+    });
+  };
   return server;
 };
 
+S3rver.defaultOptions = {
+  port: 4578,
+  hostname: 'localhost',
+  silent: false,
+  cors: fs.readFileSync(path.resolve(__dirname, '../test/resources/cors_sample_policy.xml')),
```

Comment:
I don't want to make it in-memory since I have it linked in the README. I honestly only have it there because it felt out of place throwing it in the `lib` folder. Keeping it at root level would make enough sense if the `test/resources` location seems too volatile.

Reply:
We'll keep an eye out, tests should fail if the config gets moved/deleted